tests/integration/apigateway/test_apigateway_common.py

```diff
@@ -0,0 +1,234 @@
+    requests/responses from the API.
+    """
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(
+        paths=[
+            "$.invalid-request-body.Type",
+            "$..methodIntegration.cacheNamespace",
+            "$..methodIntegration.integrationResponses",
+            "$..methodIntegration.passthroughBehavior",
+            "$..methodIntegration.requestParameters",
+            "$..methodIntegration.timeoutInMillis",
+        ]
+    )
+    def test_api_gateway_request_validator(
+        self,
+        apigateway_client,
+        create_lambda_function,
+        create_rest_apigw,
+        apigw_redeploy_api,
+        lambda_client,
+        snapshot,
+    ):
+        # TODO: create fixture which will provide basic integrations where we can test behaviour
+        # see once we have more cases how we can regroup functionality into one or several fixtures
+        # example: create a basic echo lambda + integrations + deploy stage
+        snapshot.add_transformers_list(
+            [
+                snapshot.transform.key_value("requestValidatorId"),
+                snapshot.transform.key_value("id"),  # deployment id
+                snapshot.transform.key_value("fn_name"),  # lambda name
+                snapshot.transform.key_value("fn_arn"),  # lambda arn
+            ]
+        )
+
+        fn_name = f"test-{short_uid()}"
+        create_lambda_function(
+            func_name=fn_name,
+            handler_file=TEST_LAMBDA_AWS_PROXY,
+            runtime=LAMBDA_RUNTIME_PYTHON39,
+        )
+        lambda_arn = lambda_client.get_function(FunctionName=fn_name)["Configuration"][
+            "FunctionArn"
+        ]
+        # matching on lambda id for reference replacement in snapshots
+        snapshot.match("register-lambda", {"fn_name": fn_name, "fn_arn": lambda_arn})
+
+        parsed_arn = parse_arn(lambda_arn)
+        region = parsed_arn["region"]
+        account_id = parsed_arn["account"]
+
+        api_id, _, root = create_rest_apigw(name="aws lambda api")
+
+        resource_1 = apigateway_client.create_resource(
+            restApiId=api_id, parentId=root, pathPart="test"
+        )["id"]
+
+        resource_id = apigateway_client.create_resource(
+            restApiId=api_id, parentId=resource_1, pathPart="{test}"
+        )["id"]
+
+        validator_id = apigateway_client.create_request_validator(
+            restApiId=api_id,
+            name="test-validator",
+            validateRequestParameters=True,
+            validateRequestBody=True,
+        )["id"]
+
+        apigateway_client.put_method(
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="POST",
+            authorizationType="NONE",
+            requestValidatorId=validator_id,
+            requestParameters={"method.request.path.test": True},
+        )
+
+        apigateway_client.put_integration(
```

Comment:
nit: for simple integrations, we could try leveraging and extending the `create_rest_api_with_integration` fixture. It currently only supports `dynamodb` and `kinesis` integrations, but Lambda would be a great addition.. 👍

Reply:
Makes sense! I need to take a look at it, and how we can customise it. I guess we can always create simple integrations and use `update_*` afterwards, for example the `requestParameters` in that case. There are so many parameters possible.
I'll take a good look a those once I'm starting with integrations. This PR was a bit outside my "normal" work for now, as I was fixing the change in logic now that we are raising exceptions for non-existing resources. 😄